master/internal/task/idler/watcher.go

```diff
@@ -0,0 +1,121 @@
+package idler
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/determined-ai/determined/master/pkg/ptrs"
+	"github.com/determined-ai/determined/master/pkg/syncx/waitgroupx"
+
+	log "github.com/sirupsen/logrus"
+
+	"github.com/determined-ai/determined/master/internal/sproto"
+
+	"github.com/determined-ai/determined/master/internal/proxy"
+)
+
+var syslog = log.WithField("component", "idle-watcher")
+
+// ErrIdle indicates that, according to the Watcher configuration, the service is idle.
+var ErrIdle = fmt.Errorf("service is inactive")
+
+// TickInterval is the interval at which to check the proxy activity.
+var TickInterval = 5 * time.Second
+
+// Watcher watches the proxy activity to handle a task actor idle timeout.
+type Watcher struct {
+	// System dependencies.
+	syslog *log.Entry
+
+	// Configuration.
+	cfg    *sproto.IdleTimeoutConfig
```

Comment:
`IdleTimeoutConfig` is a pretty small struct; does this have to be a pointer?

Reply:
eh, sure